InscryptionAPI/Triggers/Interfaces.cs

```diff
@@ -1,240 +1,288 @@
+        /// </summary>
+        /// <returns>Modified data about the amount of slots that will be selected.</returns>
+        public int CollectGetAttackSlotCount(PlayableCard card);
+    }
+
+    /// <summary>
+    /// Flag that tells IGetOpposingSlots when to trigger.
+    /// </summary>
+    public enum OpposingSlotTriggerPriority
+    {
+        /// <summary>
+        /// Normal mode - use this for opposing slot modifications that either don't do anything to the original opposing slots or remove it.
+        /// </summary>
+        Normal,
+        /// <summary>
+        /// Bring back opposing slot - use this for opposing slot modifications that bring back the original opposing slot if it was removed like Trifurcated Strike.
+        /// </summary>
+        BringsBackOpposingSlot,
+        /// <summary>
+        /// Post slot addition modification - use this for opposing slot modifications that don't interact with any other modification and instead just statically modify the finished result like Double Strike.
+        /// </summary>
+        PostAdditionModification,
+        /// <summary>
+        /// Replaces default opposing slot - use this for opposing slot modifications that replace the original opposing slot like Omni Strike.
+        /// </summary>
+        ReplacesDefaultOpposingSlot
+    }
+
+    /// <summary>
+    /// Data collection trigger that collects data about the usability of an item.
+    /// </summary>
+    public interface IItemCanBeUsed
+    {
+        public bool RespondsToItemCanBeUsed(string itemname, bool currentValue);
+        public bool CollectItemCanBeUsed(string itemname, bool currentValue);
+    }
+
+    /// <summary>
+    /// Trigger that is triggered when an item is prevented from use using IItemCanBeUsed.
+    /// </summary>
+    public interface IOnItemPreventedFromUse
+    {
+        public bool RespondsToItemPreventedFromUse(string itemname);
+        public IEnumerator OnItemPreventedFromUse(string itemname);
+    }
+
+    /// <summary>
+    /// Data collection trigger that modifies the passive attack buffs of any card.
+    /// </summary>
+    public interface IOnCardPassiveAttackBuffs
+    {
+        public bool RespondsToCardPassiveAttackBuffs(PlayableCard card, int currentValue);
+        public int CollectCardPassiveAttackBuffs(PlayableCard card, int currentValue);
+    }
+
+    /// <summary>
+    /// Data collection trigger that modifies the passive health buffs of any card.
+    /// </summary>
+    public interface IOnCardPassiveHealthBuffs
+    {
+        public bool RespondsToCardPassiveHealthBuffs(PlayableCard card, int currentValue);
+        public int CollectCardPassiveHealthBuffs(PlayableCard card, int currentValue);
+    }
+
+    /// <summary>
+    /// Data collection trigger that modifies the damage taken by any card.
+    /// </summary>
+    public interface ICardTakenDamageModifier
```

Comment:
Needs better documentation to describe how this interface is supposed to work.

Reply:
i guess, sure

Reply:
still need to do this

Reply:
done